sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -508,6 +508,91 @@ case class GpuArrayMax(child: Expression) extends GpuUnaryExpression with Implic
     }
 }
 
+case class GpuArrayRepeat(left: Expression, right: Expression) extends GpuBinaryExpression {
+
+  override def dataType: DataType = ArrayType(left.dataType, left.nullable)
+
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    // The primary issue of array_repeat is to workaround the null and negative count.
+    // Spark returns a null (list) when encountering a null count, while cudf::repeat simply
+    // throws an exception.
+    // Spark returns a empty list when encountering a negative count, while cudf::repeat simply
+    // throws an exception.
+
+    // Step 1. replace invalid counts
+    //  null -> 0
+    //  negative values -> 0
+    val refinedCount = withResource(GpuScalar.from(0, DataTypes.IntegerType)) { zero =>
```

Comment:
Question. Did you intend to use refinedCount with `table.repeat`? If not then this is not used until after we do the repeat, so it might be cleaner to just switch steps one and two so we calculate refinedCount closer to where it is used?  This is just a nit by the way. The code looks fine.

Reply:
Yes, I think we should use refinedCount to run `table.repeat`, because `table.repeat` can not handle count values of negative and null .